src/enzo/New_Grid_ReadGrid.C

```diff
@@ -198,15 +198,23 @@ int grid::Group_ReadGrid(FILE *fptr, int GridID, HDF5_hid_t file_id,
             ENZO_FAIL("error reading NumberOfParticles.");
     }
 
+    int ActiveParticlesExist = 0;
+
     if (fscanf(fptr, "NumberOfActiveParticles = %"ISYM"\n", &NumberOfActiveParticles) != 1) {
```

Comment:
The logic here is a bit confusing. Could it not be done like this?
```suggestion
    if (fscanf(fptr, "NumberOfActiveParticles = %"ISYM"\n", &NumberOfActiveParticles) == 1) {
      fgets(unused_string, MAX_LINE_LENGTH, fptr);
      fgets(unused_string, MAX_LINE_LENGTH, fptr);
    } else {
      NumberOfActiveParticles = 0;
    }
```

Reply:
This seems like a much cleaner implementation to me.  Thanks!